File: training/java/core/learnjava/src/com/dbconnection/DBConnection.java
package com.dbconnection;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

public class DBConnection {

	public static void main(String[] args) {
		String url = "jdbc:mysql://localhost:3306/dblearnjava"; // replace with your DB name
		String user = "root"; // your username
		String password = "Medplus@321"; // your password

		try {
			// 1. Load and register JDBC driver (optional for modern versions)
			Class.forName("com.mysql.cj.jdbc.Driver");

			// 2. Create connection
			Connection conn = DriverManager.getConnection(url, user, password);
			System.out.println("Connected to the database!");

			// 3. Create a statement
			Statement stmt = conn.createStatement();

			// 4. Execute query
			ResultSet rs = stmt.executeQuery("SELECT * FROM Employee");

			// 5. Process the result
			while (rs.next()) {
				System.out.println("name : " + rs.getString("name")+" Age : "+rs.getInt("Age"));
				// Add more columns as needed
			}
			String sql = "INSERT INTO Employee (name,Age) VALUES (?,?)";
			
			PreparedStatement statement = conn.prepareStatement(sql);
			statement.setString(1, "Ajay");
			statement.setInt(2, 42);
//			int Check = statement.executeUpdate();
//			if (Check > 0 ) {
//				System.out.println("Insertion done !!!");
//			}else {
//				System.out.println("Some Error Occured ");
//			}
			String delete = "DELETE FROM Employee where name = ?";
			PreparedStatement deleteStatement = conn.prepareStatement(delete);
			deleteStatement.setString(1, "Ajay");
			deleteStatement.executeUpdate();
			System.out.println("Deletion done !!!");

			// 6. Close resources
			rs.close();
			stmt.close();
			conn.close();

		} catch (SQLException e) {
			System.out.println("SQL Error: " + e.getMessage());
		} catch (ClassNotFoundException e) {
			System.out.println("JDBC Driver not found: " + e.getMessage());
		}

	}
}

File: Python/Day6/testing/DBConnection.py
import mysql.connector

def add_employee(age,name,salary):
    try:
        conn=mysql.connector.connect(
        host="localhost",
        user="Anand",
        password="1925112816@Aa",
        database="test"
    )
        cursor=conn.cursor()
        cursor.execute('insert into Employee(age,name,salary) values(%s,%s,%s)',(age,name,salary))
        conn.commit()
        cursor.close()
        conn.close()
        return True
    except Exception as e:
        raise e
def get_employee(id):
    try:
        conn=mysql.connector.connect(
            host="localhost",
            user="Anand",
            password="1925112816@Aa",
            database="test"
        )
        cursor=conn.cursor()
        cursor.execute(f'select * from Employee where id={id}')
        result=cursor.fetchall()
        return result
    except Exception as e:
        print(e)

get_employee(1)